### PR TITLE
[ChaosCenter]: Implement api token CRUD logic & frontend view

### PR DESCRIPTION
**Pull request by namkyu1999**
## Proposed changes
Issue No. #4023 

This PR implemented backstage-plugin phase 2. Check out this [doc](https://docs.google.com/document/d/1_ePJ36DwFrhFPhcxhxXX__yiNzW1KDL83L2lfF8oIcA/edit?usp=sharing). Once this PR is merged, we'll be able to make other plugins for Litmus.

1. Add `Bearer schema` to graphql-server and frontend.
2. Add api-token CRUD apis to auth-server.
3. Add api-token api infos to swagger-ui.
4. Add 'API Tokens' section to the settings page in frontend.

Here's a frontend changes

https://github.com/litmuschaos/litmus/assets/53862866/e6e5d2ae-b8dd-499a-85b0-dad662de6408

## Types of changes

What types of changes does your code introduce to Litmus? Put an `x` in the boxes that apply
- [x] New feature (non-breaking change which adds functionality)
- [ ] Bugfix (non-breaking change which fixes an issue)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation Update (if none of the other choices applies)

## Checklist

Put an `x` in the boxes that apply. You can also fill these out after creating the PR. If you're unsure about any of them, don't hesitate to ask. We're here to help! This is simply a reminder of what we are going to look for before merging your code.
- [x] I have read the [CONTRIBUTING](https://github.com/litmuschaos/litmus/blob/master/CONTRIBUTING.md) doc
- [x] I have signed the commit for DCO to be passed.
- [x] Lint and unit tests pass locally with my changes
- [x] I have added tests that prove my fix is effective or that my feature works (if appropriate)
- [x] I have added necessary documentation (if appropriate)

## Dependency
- Please add the links to the dependent PR need to be merged before this (if any).

### CODE REVIEW

File: chaoscenter/authentication/api/routes/user_router.go [hunk 1]
```diff
@@ -13,6 +13,9 @@ func UserRouter(router *gin.Engine, service services.ApplicationService) {
 	router.POST("/login", rest.LoginUser(service))
 	router.POST("/logout", rest.LogoutUser(service))
 	router.Use(middleware.JwtMiddleware(service))
+	router.GET("/token/:uid", rest.GetApiTokens(service))
+	router.POST("/token", rest.CreateApiToken(service))
```
Comment (Saranya-jena) [inline chaoscenter/authentication/api/routes/user_router.go:17]: can we rename it to /create_token ?

File: chaoscenter/authentication/pkg/entities/session.go [hunk 2]
```diff
@@ -6,3 +6,24 @@ type RevokedToken struct {
 	ExpiresAt int64  `bson:"expires_at"`
 	CreatedAt int64  `bson:"created_at"`
 }
+
+// ApiTokenInput struct for storing ApiTokenInput
+type ApiTokenInput struct {
+	UserID               string `json:"user_id"`
+	Name                 string `json:"name"`
+	ExpirationDateInDays int    `json:"expiration_date_in_days"`
```
Comment (Saranya-jena) [inline chaoscenter/authentication/pkg/entities/session.go:14]: does it store number of days remaining for its expiration?

File: chaoscenter/authentication/pkg/services/session_service.go [hunk 3]
```diff
@@ -76,3 +79,44 @@ func (a applicationService) GetSignedJWT(user *entities.User) (string, error) {
...
+	expiresAt := time.Now().AddDate(0, 0, request.ExpirationDateInDays).Unix()
+	claims := token.Claims.(jwt.MapClaims)
+	claims["uid"] = user.ID
+	claims["role"] = user.Role
+	claims["username"] = user.Username
+	claims["exp"] = expiresAt
+
+	tokenString, err := token.SignedString([]byte(utils.JwtSecret))
+	if err != nil {
+		log.Info(err)
```
Comment (Saranya-jena) [inline chaoscenter/authentication/pkg/services/session_service.go:95]: can you change it to log.errorf() ?

File: chaoscenter/authentication/pkg/entities/session.go (hunk 2 above)
Comment (namkyu1999) [reply chaoscenter/authentication/pkg/entities/session.go]: Yes, and I rename `expiration_date_in_days` to `days_until_expiration` to clarify.

**Review by Saranya-jena — APPROVED** [review]
LGTM 🚀

**Review by SarthakJain26 — CHANGES_REQUESTED** [review]

File: chaoscenter/authentication/api/handlers/rest/user_handlers.go [hunk 4]
```diff
@@ -423,3 +423,86 @@ func UpdateUserState(service services.ApplicationService) gin.HandlerFunc {
...
+		// Checking if user exists
+		user, err := service.GetUser(apiTokenRequest.UserID)
+		if err != nil {
+			log.Error(err)
+			c.JSON(utils.ErrorStatusCodes[utils.ErrUserNotFound], presenter.CreateErrorResponse(utils.ErrUserNotFound))
+			return
+		}
+
+		// Checking if user is deactivated
+		if user.DeactivatedAt != nil {
```
[the remark this answers is not in the document]
Comment (Saranya-jena) [reply chaoscenter/authentication/api/handlers/rest/user_handlers.go]: @namkyu1999 deactivation logic is there in litmus 3.0, however, it has some bugs that we will be fixing. You can take the requested changes in the next PR.

[the remark this answers is not in the document]
Comment (namkyu1999) [reply chaoscenter/authentication/api/handlers/rest/user_handlers.go]: Okay I will make the change in the next PR 😀

**Review by SarthakJain26 — APPROVED** [review]

**Review by amityt — APPROVED** [review]
LGTM 🚀

**Comment by Saranya-jena** [comment]
@namkyu1999 Please resolve the conflicts, will merge post that.

**Pull request merged**